src/createActionArgs.js

```diff
@@ -40,17 +40,37 @@ var createStateArg = function (action, model, isAsync, compute) {
   return state
 }
 
+var createServicesArg = function (action, services, parentKey) {
+  return Object.keys(services).reduce(function (newServices, key) {
+    if (typeof services[key] === 'function') {
+      newServices[key] = function () {
+        action.services.push({
+          name: parentKey || 'root',
+          method: key,
+          args: [].slice.call(arguments)
+        })
+        return services[key].apply(services[key], arguments)
+      }
+    } else {
+      newServices[key] = createServicesArg(action, services[key], key)
```

Comment:
Okay, been thinking. We can do what you suggest. I was just unsure how to identify what module the service belonged to when it was nested, but found an easy way to identify now. Let me do another iteration and lets have a new look :)

Wont be ready before tomorrow morning though, got the chat on Falcor and need some rest. But yeah, tomorrow morning I will have a happy solution :)